user/syscalls.c

```diff
@@ -56,9 +100,22 @@ ssize_t write(int fd, const void *buf, size_t count) {
   return retval;
 }
 
-_off_t lseek(int __fildes, _off_t __offset, int __whence) {
-  errno = ENOSYS;
-  return (off_t)-1;
+_off_t lseek(int fd, _off_t offset, int whence) {
```

Comment:
I've considered using assembly, but I'm not sure if it's the right choice. The problem is with `errno`. Newlib implements it as a preprocessor macro, which expands to `(*__errno())`. The function `__errno` ensures reentrancy and returns an indirect pointer to where error value should be written. So that easily works in C, but not in assembly. I might reproduce this behavior in assembly, but it will make us incompatible with other standard libraries, and I am not sure whether we with to commit to newlib entirely. I am also not sure if a differently configured newlib won't change how `errno` is implemented.